### PR TITLE
Use Task Manager, update timezone data

### PR DESCRIPTION
**Pull request by rbergen**
## Description

This:

- Employs the task manager for managing tasks that were started separately up to this point
- Updates the timezone data, and includes a script to generate it
- (Re)introduces mist as a recognized condition in the Weather effect
- Refactors the settings implementation and introduces device-specific settings
- Removes the Brooklyn Room project from the web projects config

## Contributing requirements

* [x] I read the contribution guidelines in [CONTRIBUTING.md](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/CONTRIBUTING.md).
* [x] I understand the BlinkenPerBit metric, and maximized it in this PR.
* [x] I selected `main` as the target branch.
* [x] All code herein is subjected to the license terms in [COPYING.txt](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/COPYING.txt).

### CODE REVIEW

File: include/effectmanager.h [hunk 1]
```diff
@@ -202,19 +198,23 @@ class EffectManager : public IJSONSerializable
...
             return false;
 
-        _abEffectEnabled = std::make_unique<bool[]>(_vEffects.size());
-
-        // Try to load effect enabled state from JSON also, default to "enabled" otherwise
-        JsonArrayConst enabledArray = jsonObject["eef"].as<JsonArrayConst>();
-        int enabledSize = enabledArray.isNull() ? 0 : enabledArray.size();
-
-        for (int i = 0; i < _vEffects.size(); i++)
+        if (jsonObject.containsKey("eef"))
```
Comment (davepl) [inline include/effectmanager.h:201]: Might be worth a comment what eef ivl and so on are...

File: include/ledstripeffect.h [hunk 2]
```diff
@@ -48,9 +51,23 @@ class LEDStripEffect : public IJSONSerializable
     size_t _cLEDs;
     String _friendlyName;
     int    _effectNumber;
+    bool   _enabled = true;
+    std::vector<SettingSpec> _settingSpecs;
 
     std::shared_ptr<GFXBase> _GFX[NUM_CHANNELS];
 
+    template<typename Tv>
+    bool SetIfSelected(String settingName, const String& propertyName, Tv& property, const Tv& value)
```
Comment (davepl) [inline include/ledstripeffect.h:60]: Does settingName need to be passed by value?

**Review by davepl — APPROVED** [review]
A few minor comments but no bugs to be found!

File: include/effectmanager.h (hunk 1 above)
Comment (rbergen) [reply include/effectmanager.h:201]: Fair point, will add one.

File: include/effectmanager.h [hunk 3]
```diff
@@ -382,9 +376,11 @@ class EffectManager : public IJSONSerializable
             return;
         }
 
-        if (!_abEffectEnabled[i])
+        auto effect = _vEffects[i];
```
[the remark this answers is not in the document]
Comment (rbergen) [reply include/effectmanager.h:379]: Good point, I will change this.

File: include/ledstripeffect.h (hunk 2 above)
Comment (rbergen) [reply include/ledstripeffect.h:60]: Instinctively I'd say no, but I also find it difficult believe I randomly left one parameter unreferenced. I'll check and modify if appropriate.

File: src/jsonserializer.cpp [hunk 4]
```diff
@@ -82,27 +83,32 @@ bool LoadJSONFile(const char *fileName, size_t& bufferSize, std::unique_ptr<Allo
...
             break;
 
         pJsonDoc.reset(nullptr);
         bufferSize += JSON_BUFFER_INCREMENT;
 
         debugW("Out of memory serializing object - increasing buffer to %zu bytes", bufferSize);
     }
+}
+
+bool SaveToJSONFile(const char *fileName, size_t& bufferSize, IJSONSerializable& object)
```
[the remark this answers is not in the document]
Comment (rbergen) [reply src/jsonserializer.cpp:104]: Yes, and it soon will be. :) I saw you made this switch in your upcoming PR, so I decided not to bother making the same change again.

File: include/ledstripeffect.h (hunk 2 above)
Comment (rbergen) [reply include/ledstripeffect.h:60]: I checked, and I must not have had my morning coffee when I wrote that. There is no reason not to const reference that.
Good catch!

**Pull request merged**